### PR TITLE
phonenumbers: delete phone number from cache when user deletes it

### PR DESCRIPTION
**Pull request by jakob223**
cc @patrickxb @keybase/y2ksquad 

Issue: Y2K-391

### CODE REVIEW

**Comment by heronhaye** [comment]
sorry i'll leave one comment: add a test please

File: go/phonenumbers/user.go [hunk 1]
```diff
@@ -72,6 +73,11 @@ func DeletePhoneNumber(mctx libkb.MetaContext, phoneNumber keybase1.PhoneNumber)
 	}
 
 	_, err := mctx.G().API.Delete(mctx, arg)
+
+	if err != nil {
```
Comment (patrickxb) [inline go/phonenumbers/user.go]: if err == nil?

File: go/contacts/phonebook.go [hunk 2]
```diff
@@ -92,3 +92,21 @@ func (s *SavedContactsStore) RetrieveContacts(mctx libkb.MetaContext) (ret []key
...
 }
+
+func (s *SavedContactsStore) ClearPhoneNumber(mctx libkb.MetaContext,
+	phoneNumber keybase1.PhoneNumber) {
+	contactList, err := s.RetrieveContacts(mctx)
+	if err != nil {
+		mctx.Warning("Failed to get cached contact list: %x", err)
+	}
+	for i, con := range contactList {
+		if *con.Component.PhoneNumber == keybase1.RawPhoneNumber(phoneNumber) {
```
Comment (zapu) [inline go/contacts/phonebook.go]: `con.Component.PhoneNumber` can be nil if it's an email component

**Comment by zapu** [comment]
@heronhaye can you take a look? this is still pretty rough but might be good enough for now

File: go/emails/user.go [hunk 3]
```diff
@@ -77,7 +91,13 @@ func SetVisibilityEmail(mctx libkb.MetaContext, email keybase1.EmailAddress, vis
 	}
 
 	_, err := mctx.G().API.PostJSON(mctx, arg)
-	return err
+	if err != nil {
+		return err
+	}
+	if visibility == keybase1.IdentityVisibility_PRIVATE {
```
Comment (heronhaye) [inline go/emails/user.go]: good catch, also need to do it below for SetVisibilityAllEmail

Comment (heronhaye) [reply go/emails/user.go]: (same for phone)

File: go/contacts/phonebook.go [hunk 4]
```diff
@@ -92,3 +92,40 @@ func (s *SavedContactsStore) RetrieveContacts(mctx libkb.MetaContext) (ret []key
...
+		mctx.Warning("Failed to get cached contact list: %x", err)
+		return
+	}
+	for i, con := range contactList {
+		var unresolve bool
+		switch {
+		case phoneNumber != nil && con.Component.PhoneNumber != nil:
+			unresolve = *con.Component.PhoneNumber == keybase1.RawPhoneNumber(*phoneNumber)
+		case email != nil && con.Component.Email != nil:
+			unresolve = *con.Component.Email == *email
```
Comment (heronhaye) [inline go/contacts/phonebook.go]: could quickly change this to use `strings.ToLower`

**Pull request merged**